From 20450e28990a596f28d1e3167e11d579f435db38 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Wed, 20 Sep 2023 09:49:29 +0200
Subject: [PATCH 05/17] makedep: Allow unixlibs to have a different 64bit name.

CW-Bug-Id: #22729
(cherry picked from commit bc75d41ceea1607893dc0c4a32fadc785abcc55b)
---
 tools/makedep.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/tools/makedep.c b/tools/makedep.c
index a15e0e98b41..bababcf36ae 100644
--- a/tools/makedep.c
+++ b/tools/makedep.c
@@ -4082,12 +4082,14 @@ static void output_unix_lib( struct makefile *make )
     struct strarray unix_deps = empty_strarray;
     struct strarray unix_libs = add_unix_libraries( make, &unix_deps );
     unsigned int arch = 0;  /* unix libs are always native */
+    const char *unixlib;
 
     if (make->disabled[arch]) return;
+    if (!(unixlib = get_expanded_make_variable( make, strmake( "%s_%s", archs.str[arch], "UNIXLIB" ) ))) unixlib = make->unixlib;
 
-    strarray_add( &make->all_targets[arch], make->unixlib );
-    install_program( make, make->unixlib, arch, make->unixlib, arch_install_dirs[arch] );
-    output( "%s:", obj_dir_path( make, make->unixlib ));
+    strarray_add( &make->all_targets[arch], unixlib );
+    install_program( make, unixlib, arch, unixlib, arch_install_dirs[arch] );
+    output( "%s:", obj_dir_path( make, unixlib ));
     output_filenames_obj_dir( make, make->unixobj_files );
     output_filenames( unix_deps );
     output( "\n" );
-- 
2.55.0

